packages/autorest.python/autorest/preprocess/__init__.py

```diff
@@ -135,6 +135,37 @@ def add_overloads_for_body_param(yaml_data: Dict[str, Any]) -> None:
     content_type_param["optional"] = True
 
 
+def add_overloads_for_native_overloads(
```

Comment:
why do we need different logic for cadl natively defined overloads and the overloads we're adding?

Reply:
@iscai-msft similar reason with https://github.com/Azure/autorest.python/pull/1770#discussion_r1127292627 that I don't want to break previous logic and it is very complicated if trying to merge the logic together. In view of the complexity and maintenance, I think it is better to split the logic.